src/Utility/Filter.php

```diff
@@ -0,0 +1,126 @@
+<?php
+declare(strict_types=1);
+
+/**
+ * CakePHP(tm) : Rapid Development Framework (https://cakephp.org)
+ * Copyright (c) Cake Software Foundation, Inc. (https://cakefoundation.org)
+ *
+ * Licensed under The MIT License
+ * For full copyright and license information, please see the LICENSE.txt
+ * Redistributions of files must retain the above copyright notice.
+ *
+ * @copyright     Copyright (c) Cake Software Foundation, Inc. (https://cakefoundation.org)
+ * @since         5.0.1
+ * @license       https://opensource.org/licenses/mit-license.php MIT License
+ */
+
+namespace Cake\Utility;
+
+use JsonException;
+use Stringable;
+
+/**
+ * Methods for converting mixed values to specific data types, ensuring a type-safe approach to data manipulation.
+ * This utility is useful for safely narrowing down the data types.
+ */
+class Filter
+{
+    /**
+     * Converts the given value to a string.
+     *
+     * This method attempts to convert the given value to a string.
+     * If the value is already a string, it returns the value as it is.
+     * If the conversion is not possible, it returns NULL.
+     *
+     * @param mixed $value The value to be converted.
+     * @return ?string Returns the string representation of the value, or null if the value is not a string.
+     */
+    public static function toString(mixed $value): ?string
+    {
+        if (is_string($value)) {
+            return $value;
+        } elseif (is_int($value)) {
+            return (string)$value;
+        } elseif (is_bool($value)) {
+            return $value ? '1' : '0';
```

Comment:
You could skip the `elseif` here as each case returns :golfing_man:

Reply:
thx. Fixed in https://github.com/cakephp/cakephp/pull/17295/commits/7f0d0282ade2adde17154d8b30d872f568372224